Makefile

```diff
@@ -105,12 +106,12 @@ doc: clean
 	cargo doc --no-deps --open -v
 
 edit:
-	$EDITOR src/lib.rs &
-	godot -e &
+	$$EDITOR rust/src/lib.rs &
+	godot {{godot_project_path_arg}} -e &
```

Comment:
Extra `$` here? As a side note I'm not really sure if the editor should be opened here: `$EDITOR` might be pointing to a CLI editor like Vim or Emacs, and opening one of those in the same terminal as Godot is asking for trouble.

Reply:
done...

>  As a side note I'm not really sure if the editor should be opened here: $EDITOR might be pointing to a CLI editor like Vim or Emacs, and opening one of those in the same terminal as Godot is asking for trouble.

we could indicate in the wiki that it should be a gui based editor and not a cli editor, or they could just replace `$EDITOR` with their text editor of choice

Reply:
I'm just not sure how universally applicable that line is. Some GUI editors don't work very well with that command, e.g. VS Code, which works best when given a path to a workspace (`rust`). My point is that since it doesn't necessarily produce desirable effects most of the time, it should not be done by default.

I'm certainly used to having `$EDITOR` point to a CLI editor myself, because that makes programs that use them (mostly `git`) work more smoothly, so I might be biased, but it is uncommon for GUI applications to use `$EDITOR` -- they usually prefer desktop file association.

Reply:
how about using `$VISUAL` instead?

Reply:
The "visual" in `$VISUAL` isn't referring to modern GUI, but "visual" as opposed to "printed on paper" i.e. teletypes. It's technically *the* environment variable for terminal editors, to even a larger extent than `$EDITOR`.

Reply:
Traditionally `$EDITOR` and `$VISUAL` was thought of that way, I initially thought in modern times it changed. Any suggestion on how to open the godot editor and a text editor/ide with a single command? or just comment out the `$EDITOR` line and let the user decide if he/she want to use it?

Reply:
There is `xdg-open` which will use desktop integration, which usually means a GUI editor. However, as mentioned before, `xdg-open`ing `rust/src/lib.rs` still won't work well with VSCode, which is *immensely* popular. I'd prefer if `make edit` is defined as a command that just opens the Godot editor, but commenting the line out would also be fine by me.

Reply:
done, commented it out...